handle_http.go

```diff
@@ -438,23 +438,39 @@ func parseTransferStatus(status string) (int, string) {
 	return statusCode, strings.TrimSpace(parts[1])
 }
 
+func setupTransport() http.Transport {
+	//Getting timeouts and other information from defaults.yaml
+	maxIdleConns := viper.GetInt("Transport.MaxIdleIcons")
+	idleConnTimeout := viper.GetDuration("Transport.IdleConnTimeout")
+	transportTLSHandshakeTimeout := viper.GetDuration("Transport.TLSHandshakeTimeout")
+	expectContinueTimeout := viper.GetDuration("Transport.ExpectContinueTimeout")
+	responseHeaderTimeout := viper.GetDuration("Transport.ResponseHeaderTimeout")
+
+	transportDialerTimeout := viper.GetDuration("Transport.Dialer.Timeout")
+	transportKeepAlive := viper.GetDuration("Transport.Dialer.KeepAlive")
+
+	//Set up the transport
+	transport := http.Transport {
```

Comment:
From the documentation page for transports:

> Clients and Transports are safe for concurrent use by multiple goroutines and for efficiency should only be created once and re-used.

So let's create this as a global (using `sync.Once`; see https://github.com/PelicanPlatform/pelican/blob/main/namespace-registry/registry.go#L126-L134) and return a pointer to the global.

Reply:
Fixed